api/v1alpha1/clusterextension_types.go

```diff
@@ -78,6 +97,10 @@ type ClusterExtensionSpec struct {
 	// the bundle may contain resources that are cluster-scoped or that are
 	// installed in a different namespace. This namespace is expected to exist.
 	InstallNamespace string `json:"installNamespace"`
+
+	//+optional
+	// RegistryTLS defines the connection parameters to retrieve an image from a registry
+	RegistryTLS *ClusterExtensionTLS `json:"registryTLS,omitempty"`
```

Comment:
It feels like it does not belong to `ClusterExtension`. At least in the use case when we install packages from a catalog it feels like specifying this every time we create a `ClusterExtension` is excessive.

We need to think about UX for the following use cases:
1. We install extensions from a catalog. I think in this case we probably don't want users to specify certs in `ClusterExtension` every time they install something from a catalog. Feels like some configuration should be done on `ClusterCatalog` level once and re-used.
2. We install something directly (like future support for direct helm installation).

Reply:
Yes, this was the intent.  This is the certificate used to verify the connection to the _image-registry_ from which we get the bundle. The certificate can't be part of the catalog, as it may expire, and it could be an external repository or other location as you point out.

HOWEVER, this is _optional_, as the operator-controller will load the system certificates, and if the image registry's certificate is signed by a system certificate, then it's not necessary.

Reply:
The more I think about this - the more I think we should not have this on `ClusterExtension`. Another reason - we do not specify specific catalog in `ClusterExtension`. If on `ClusterExtension` you say you want to install `argocd-operator` - it can come from any catalog installed on the cluster: it might come from a catalog which uses an image registry where certs are issued by a public CA, but it might also come from a catalog which uses an image registry with a self-signed cert. We don't know on the `ClusterExtension` level.

I can totally see that some users are not going to know what to specific in `registryTLS` because they don't know how `argocd-operator` will resolve.

> This is the certificate used to verify the connection to the _image-registry_ from which we get the bundle.

I understand that. But whoever configures `ClusterCatalog` probably knows better about what image registries the catalog uses. So they can provide a ref to a relevant CA on `ClusterCatalog` level.

> The certificate can't be part of the catalog, as it may expire, and it could be an external repository or other location as you point out.

Not sure that I'm following this.

Reply:
If the certificate is part of the catalog metadata, it could easily become stale - especially if a separate image-registry is used. And as you point out, the argocd could come from any catalog, and subsequently any docker-registry.

Again, this is _optional_, and is intended for those specific docker-registries which may be private (e.g. our e2e docker-registries) or for which don't use a system certificate.

_This should really be considered an exception, not the rule._

@joelanford, I believe it was your idea to add a certificate to the ClusterExtension, do you have an additional response?

Reply:
> If the certificate is part of the catalog metadata

I'm not suggesting to put a certificate into catalog metadata. I'm suggesting to _potentially_ do what you did for `ClusterExtension`, but for `catalogd`'s `ClusterCatalog` API instead.

My idea is that people using `ClusterExtension` should not be forced to specify CA every time they install something on a cluster where self-signed certs are used. The cluster admins should be able to specify refs to secrets containing required CAs once (when adding a catalog to their cluster) and `ClusterExtension` API can remain the same.

What I suggest might also not be the right approach, but it is clear to me that we as a group need to spend some more time on the design for this.

> This should really be considered an exception, not the rule.

If we provide this API - we and our users will be stuck with it for long time. We will have to support it and users will be forced to use it.

I agree with you - it won't affect people who use image registry with certs signed by public CAs. But it will result in a terrible user experience for people who install something on a cluster with self-signed certs. For them I see at least the following UX issues:

1. They will have to specify `registryTLS` for every `ClusterExtension`.
2. It is not clear what CA to specify on `ClusterExtension`. You have to know in advance how your package & version are going to resolve (is it going to use a registry with self-signed certs?). I think it is going to be more of a pain if people using `ClusterExtension` are not the same people configuring catalogs on the cluster.

IMO - we should either not support this use case at all or design so that we provide nice UX for all the use cases.

Let me know if you want to discuss this on a call. We should probably invite @joelanford as the author of the original idea.

Reply:
I spoke with Joe Tuesday night and we are going to abandon the addition to the API _for now_ and re-evaluate it later. This PR will still remove the annotations and fix the e2e to properly use certificates (and not use `InsecureSkipTLSVerify`).

Reply:
Sounds good!

> This PR will still remove the annotations and fix the e2e to properly use certificates

Could you please create a new PR for this instead? This one has quite a lot comments about the original `registryTLS` implementation. I think it will be easier to review in a fresh PR.